src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -56769,6 +56769,80 @@ static void F(object? x)
                 Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(object)x").WithLocation(8, 13));
         }
 
+        [Fact]
+        public void InheritNullabilityOfNonNullableClassMember()
+        {
+            var source =
+@"#pragma warning disable 8618
+class C<T>
+{
+    internal T F;
+}
+class Program
+{
+    static void F0(string? s)
+    {
+        var a1 = new C<string>() { F = s };
+        _ = a1.F/*T:string?*/;
+        var b1 = a1;
+        _ = b1.F/*T:string!*/;
```

Comment:
>_ = b1.F/*T:string!*/; [](start = 8, length = 22)

This feels unexpected. I would expect nullability of fields to flow through an assignment. Please add a an attempt to use this field in an unsafe manner.

Reply:
I agree, nullability of class members should be copied. Logged [https://github.com/dotnet/roslyn/issues/31395](https://github.com/dotnet/roslyn/issues/31395).

---
In reply to: [236795755](https://github.com/dotnet/roslyn/pull/31392#discussion_r236795755) [](ancestors = 236795755)